### PR TITLE
Implement base instance of interface GremlinSource

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
* As the base class of Vertex, Edge and Graph.

Signed-off-by: Pan Li <panli@microsoft.com>

### CODE REVIEW

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/BasicGremlinSource.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        this.properties = new HashMap<>();
+        this.script = null;
+    }
+
+    public void setGremlinScriptStrategy(@NonNull GremlinScript script) {
+        this.setGremlinScriptStrategy(script);
+    }
+
+    public Object generateGremlinScript() {
+        return this.script.generateScript(this);
```
Comment (sophiaso) [inline src/main/java/com/microsoft/spring/data/gremlin/conversion/BasicGremlinSource.java]: Dunno what is this generated script, so it is impossible to narrow the type here?

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/BasicGremlinSource.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    private boolean hasProperty(String key) {
+        return this.properties.get(key) != null;
+    }
+
+    @Override
+    public void setProperty(String key, Object value) {
+        if (this.hasProperty(key) && value == null) {
+            this.properties.remove(key);
```
Comment (sophiaso) [inline src/main/java/com/microsoft/spring/data/gremlin/conversion/BasicGremlinSource.java:49]: Property value cannot be changed once set?

Comment (sophiaso) [reply src/main/java/com/microsoft/spring/data/gremlin/conversion/BasicGremlinSource.java:49]: Sorry, my mistake, please ignore.

**Review by sophiaso — APPROVED** [review]

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/BasicGremlinSource.java (hunk 2 above)
Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/spring/data/gremlin/conversion/BasicGremlinSource.java:49]: never mind 💯

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/BasicGremlinSource.java (hunk 1 above)
Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/spring/data/gremlin/conversion/BasicGremlinSource.java]: make sense. Thanks, and I will add generic to class.

**Pull request merged**